daemon/container_linux.go

```diff
@@ -999,7 +1005,7 @@ func (container *Container) getNetworkedContainer() (*Container, error) {
 }
 
 func (container *Container) ReleaseNetwork() {
-	if container.hostConfig.NetworkMode.IsContainer() || container.daemon.config.DisableNetwork {
+	if container.hostConfig.NetworkMode.IsContainer() {
```

Comment:
You still need to return here if `container.Config.NetworkDisabled` is `true` so that we don't attempt to release network on a container which has it's networking disabled.